scipy/stats/_distn_infrastructure.py

```diff
@@ -3855,3 +3873,437 @@ def get_distribution_names(namespace_pairs, rv_base_class):
+    success : bool or None
+        Whether the optimizer considered the optimization to terminate
+        successfully or not.
+    message : str or None
+        Any status message provided by the optimizer.
+
+    """
+
+    def __init__(self, dist, data, discrete, nllf, res):
+        self.dist = dist
+        self.data = data
+        self.discrete = discrete
+        self.pxf = getattr(dist, "pmf", None) or getattr(dist, "pdf", None)
+        self.success = getattr(res, "success", None)
+        self.message = getattr(res, "message", None)
+
+        shape_names = [] if dist.shapes is None else dist.shapes.split(", ")
+        if not discrete:
+            FitParams = namedtuple('FitParams', shape_names + ['loc', 'scale'])
+        else:
+            FitParams = namedtuple('FitParams', shape_names + ['loc'])
+
+        self.params = FitParams(*res.x)
+        self.objective_val = getattr(res, "fun", None)
+        if self.objective_val is None:
+            self.objective_val = nllf(res.x)
+        self.nllf = nllf
+
+    def __repr__(self):
+        keys = ["dist", "data", "nllf", "params", "objective_val",
+                "success", "message"]
+        m = max(map(len, keys)) + 1
+        return '\n'.join([key.rjust(m) + ': ' + repr(getattr(self, key))
+                          for key in keys if getattr(self, key) is not None])
+
+    def plot(self, ax=None):
+        """Visualize the fit result.
+
+        Superposes the PDF/PMF of the fitted distribution over a normalized
+        histogram of the data.
+
+        Available only if ``matplotlib`` is installed.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes
+            Axes object to draw the plot onto, otherwise uses the current Axes.
+
+        Returns
+        -------
+        ax : matplotlib.axes.Axes
+            The matplotlib Axes object on which the plot was drawn.
+        """
+        try:
+            from matplotlib.ticker import MaxNLocator  # type: ignore[import]
+        except ModuleNotFoundError:
+            message = "matplotlib must be installed to use method `plot`."
+            raise ValueError(message)
+
+        if ax is None:
+            import matplotlib.pyplot as plt  # type: ignore[import]
+            ax = plt.gca()
+
+        fit_params = np.atleast_1d(self.params)
+        support = self.dist.support(*fit_params)
+        lb = support[0] if np.isfinite(support[0]) else min(self.data)
+        ub = support[1] if np.isfinite(support[1]) else max(self.data)
+
+        if self.discrete:
+            x = np.arange(lb, ub + 2)
+            y = self.pxf(x, *fit_params)
+            ax.vlines(x[:-1], 0, y[:-1], label='Fit Distribution PMF',
+                      color='C0')
+            options = dict(density=True, bins=x, align='left', color='C1')
+            ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+
+        else:
+            x = np.linspace(lb, ub, 200)
```

Comment:
Perhaps. I imagine we can find degenerate cases in which either choice will look weird. Rationalizing why I'd prefer using the bounds of the data, I think it's because this plot is supposed to show how well the curve fits the data, not to help the user investigate the fit distribution. The concern with using `x = np.linspace(self.ppf(0.01), self.ppf(0.99), 100)` is that this could be much smaller or much bigger than desired. I think `x = np.linspace(min(self.ppf(0.01), data.min()), max(self.ppf(0.99), data.max(), 100)` would be better, but still could be zoomed out far away from the data. But I don't care very much either way. Shall we see if there are other, stronger opinions?